database/src/database.rs

```diff
@@ -43,11 +41,8 @@ pub trait DatabaseRo<'tx, T: Table> {
     /// from the iterator.
     fn get_range<'a, Range>(
         &'a self,
-        range: &'a Range,
-    ) -> Result<
-        impl Iterator<Item = Result<impl ValueGuard<T::Value>, RuntimeError>> + 'a,
-        RuntimeError,
-    >
+        range: Range,
+    ) -> Result<impl Iterator<Item = Result<T::Value, RuntimeError>> + 'a, RuntimeError>
```

Comment:
For `get_range()`, the iterator itself is tied to `self` and `range` but the values actually yielded from the iterator are fully owned, which is tested...